picotest_macros/src/lib.rs

```diff
@@ -41,147 +64,51 @@ pub fn picotest(attr: TokenStream, item: TokenStream) -> TokenStream {
+                    } else {
+                        item
                     }
-                    Item::Use(use_stmt) => {
-                        if traverse_use_item(&use_stmt.tree, vec!["std", "sync", "OnceLock"])
-                            .is_some()
-                        {
-                            has_once_lock = true;
-                        }
-                        if traverse_use_item(
-                            &use_stmt.tree,
-                            vec!["std", "sync", "atomic", "AtomicUsize"],
-                        )
-                        .is_some()
-                        {
-                            use_atomic_usize = true;
-                        }
-                        if traverse_use_item(
-                            &use_stmt.tree,
-                            vec!["std", "sync", "atomic", "Ordering"],
-                        )
-                        .is_some()
-                        {
-                            use_atomic_ordering = true;
-                        }
-                        if traverse_use_item(&use_stmt.tree, vec!["std", "panic"]).is_some() {
-                            use_panic = true;
-                        }
-                        Item::Use(use_stmt)
-                    }
-                    e => e,
                 })
                 .collect();
 
-            let mut use_content = vec![parse_quote!(
+            let mut content = vec![parse_quote!(
                 use picotest::*;
             )];
-            if !has_once_lock {
-                use_content.push(parse_quote!(
-                    use std::sync::OnceLock;
-                ));
-            }
-
-            if !use_atomic_usize {
-                use_content.push(parse_quote!(
-                    use std::sync::atomic::AtomicUsize;
-                ));
-            }
-            if !use_atomic_ordering {
-                use_content.push(parse_quote!(
-                    use std::sync::atomic::Ordering;
-                ));
-            }
-
-            if !use_panic {
-                use_content.push(parse_quote!(
-                    use std::panic;
-                ));
-            }
-
-            use_content.push(parse_quote!(
-                static CLUSTER: OnceLock<Cluster> = OnceLock::new();
+            content.push(parse_quote!(
+                use std::panic;
             ));
-            use_content.append(&mut e);
-
-            let mut tear_down_fn = vec![parse_quote! {
-                #[ctor::dtor]
-                fn tear_down() {
-                    if let Some(cluster) = CLUSTER.get() {
-                        cluster.stop();
+            content.append(&mut items);
+            if scope == Scope::Module {
+                let mut module_fixture: Vec<Item> = vec![parse_quote! {
+                    #[fixture]
+                    #[once]
+                    pub fn cluster(
+                        #[default(".")] plugin_path: &str,
```

Comment:
Здесь тоже дефолт учитываем?
```suggestion
                        #[default(#path)] plugin_path: &str,
```

Reply:
не сработает, данная фикстура не находится под макросом, можно лишь отпилить дефолт оставив это на усмотрение макроса

Reply:
Слушай, я попробовал подставить `#path` и `#timeout_secs` туда и вроде как он верно код генерирует. У меня вот такое получилось:
```rust
pub fn default() -> &'static &'static Cluster {
            let plugin_path = {
                use rstest::magic_conversion::*;
                (&&&Magic::<&str>(core::marker::PhantomData))
                    .magic_conversion("../tmp/test_plugin")
            };
            let timeout = 5u64;
            Self::get(plugin_path, timeout)
}
```

Reply:
все работает, если ты обратишь внимание как переделывается функция, там будет явная передача path и timeout в фикстуру

Reply:
Так, я тогда завтра еще раз отсмотрю все. Эти комменты уже не актуальны получается.